alternator/consumed_capacity.cc

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2024-present ScyllaDB
+ */
+
+/*
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+#include "consumed_capacity.hh"
+#include <seastar/core/metrics.hh>
+#include "log.hh"
+
+namespace alternator {
+
+static uint64_t MULTIPLIER = 2ULL;
+static uint64_t KB = 1024ULL;
+consumed_capacity::consumed_capacity() {
+    _metrics.add_group("alternator", {
+                   seastar::metrics::make_counter("rcu_total", seastar::metrics::description("the total RCU consumed"), _rcu).set_skip_when_empty(),
+                   seastar::metrics::make_counter("wcu_total", seastar::metrics::description("the total WCU consumed"), _wcu).set_skip_when_empty()
+    });
+}
+
+uint64_t add_consumption_helper::get_units(uint64_t length) const noexcept {
+    uint64_t bytes = MULTIPLIER*((_units + length -1)/length);
+
+    if (_isTransaction) {
+        bytes *= 2;
+    }
+    if (_isQuorum) {
+        bytes /= 2;
+    }
+    return bytes;
+}
+
+add_consumption_helper& add_consumption_helper::operator +=(uint64_t units) {
+    _units += units;
+    return *this;
+}
+
+double add_consumption_helper::get_consumptions() const noexcept {
+    return get_units((_rcu)? 4*KB : KB)/static_cast<double>(MULTIPLIER);
+}
+
+
+add_consumption_helper consumed_capacity::add_consumption(bool should_add, bool rcu, bool isQuorum, bool isTransaction) {
+    return add_consumption_helper(this, should_add, rcu, isQuorum, isTransaction);
+}
+
+add_consumption_helper::~add_consumption_helper() {
+    if (!_consumed_capacity) {
+        return;
+    }
+    if (_rcu) {
+        _consumed_capacity->_rcu += get_units(4*KB);
+    } else {
+        _consumed_capacity->_wcu += get_units(KB);
```

Comment:
```suggestion
        _consumed_capacity->_wcu += get_units(1*KB);
```

(for easier readability)